test-magnolia/src/main/scala-3/zio/test/magnolia/TypeUnionDerivation.scala

```diff
@@ -0,0 +1,65 @@
+package zio.test.magnolia
```

Comment:
I haven't read through the code too closely, but do we need this to be in the `magnolia` package? Afaict we're not using magnolia here (or maybe we are and I missed it, in that case please ignore me)

Reply:
We're not using Magnolia here indeed. Where should I put this?

Reply:
@kyri-petrou Any change to get this merged, please? I need it at work

Reply:
Sorry I'm getting way too many github notifications and I missed this. Merging now :)